Highway.cpp

```diff
@@ -13,23 +16,45 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     do not call `setSpeed`.  Pick a different function.
     */
+
+    if (Car* car = dynamic_cast<Car*>(v); car != nullptr)
+    {
+        car->closeWindows();
+    }
+    else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v); motorcycle != nullptr)
+    {
+        motorcycle->lanesplitAndRace(130);
+    }
+    else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+    {
+        semiTruck->headlightFlasher();
+    }
 }
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that tries to evade the cops. 
     do not call `setSpeed`.  Pick a different function.
     trucks pull over, but cars and bikes try to evade!!
     */
+
+    if (Car* car = dynamic_cast<Car*>(v); car != nullptr)
+    {
+        car->tryToEvade();
+    }
+    else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v); motorcycle != nullptr)
+    {
+        motorcycle->tryToEvade();
+    }
+    else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+    {
+        semiTruck->pullOver();
+    }
```

Comment:
- use `auto*`.
- you don't need to explicitly check for `nullptr`.  

`nullptr` is effectively `0`, and the computer interprets `0` as `false`.  
if the `dynamic_cast` fails, the pointer will be `nullptr`, which evaluates in this context to `false`

- [ ] revise